File: output/llama_4_maverick/avatar/algo_based_translation/java/python/codeforces_7_B.py
def main():
    t, m = map(int, input().split())
    memory = [0] * m
    allocIdx = 0

    for _ in range(t):
        command = input().strip()
        if command == "alloc":
            n = int(input())
            length = 0
            canAlloc = False
            start_idx = 0
            for i in range(m):
                if memory[i] == 0:
                    length += 1
                else:
                    length = 0
                if length == n:
                    canAlloc = True
                    start_idx = i - n + 1
                    break
            if canAlloc:
                allocIdx += 1
                for i in range(start_idx, start_idx + n):
                    memory[i] = allocIdx
                print(allocIdx)
            else:
                print("NULL")
        elif command == "erase":
            x = int(input())
            if x <= 0:
                print("ILLEGAL_ERASE_ARGUMENT")
            else:
                hasErased = False
                for i in range(m):
                    if memory[i] == x:
                        memory[i] = 0
                        hasErased = True
                if not hasErased:
                    print("ILLEGAL_ERASE_ARGUMENT")
        elif command == "defragment":
            d = 0
            for i in range(m):
                if memory[i] == 0:
                    d += 1
                else:
                    memory[i - d] = memory[i]
            for i in range(m - d, m):
                memory[i] = 0
        else:
            print("h")

if __name__ == "__main__":
    main()

File: output/deepseek_r1/avatar/algo_based_translation/python/java/codeforces_7_B.java

import java.util.ArrayList;
import java.util.Scanner;

public class codeforces_7_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] initParams = scanner.nextLine().split(" ");
        int commandCount = Integer.parseInt(initParams[0]);
        int memorySize = Integer.parseInt(initParams[1]);

        ArrayList<Integer> memory = new ArrayList<>();
        for (int i = 0; i < memorySize; i++) {
            memory.add(0);
        }
        int currentId = 1;

        for (int i = 0; i < commandCount; i++) {
            String[] command = scanner.nextLine().split(" ");
            switch (command[0]) {
                case "alloc":
                    int blockSize = Integer.parseInt(command[1]);
                    boolean allocated = false;
                    int freeCounter = 0;

                    for (int j = 0; j < memorySize; j++) {
                        if (memory.get(j) == 0) {
                            freeCounter++;
                            if (freeCounter == blockSize) {
                                int startIndex = j - blockSize + 1;
                                for (int k = startIndex; k <= j; k++) {
                                    memory.set(k, currentId);
                                }
                                System.out.println(currentId);
                                currentId++;
                                allocated = true;
                                break;
                            }
                        } else {
                            freeCounter = 0;
                        }
                    }
                    if (!allocated) {
                        System.out.println("NULL");
                    }
                    break;

                case "erase":
                    int targetId = Integer.parseInt(command[1]);
                    if (targetId == 0 || !memory.contains(targetId)) {
                        System.out.println("ILLEGAL_ERASE_ARGUMENT");
                    } else {
                        for (int j = 0; j < memorySize; j++) {
                            if (memory.get(j).equals(targetId)) {
                                memory.set(j, 0);
                            }
                        }
                    }
                    break;

                case "defragment":
                    ArrayList<Integer> compacted = new ArrayList<>();
                    int zeroCount = 0;
                    for (int block : memory) {
                        if (block != 0) {
                            compacted.add(block);
                        } else {
                            zeroCount++;
                        }
                    }
                    for (int j = 0; j < zeroCount; j++) {
                        compacted.add(0);
                    }
                    memory = compacted;
                    break;
            }
        }
        scanner.close();
    }
}

